auth/src/main/java/com/firebase/ui/auth/AuthUI.java

```diff
@@ -685,13 +982,35 @@ private SignInIntentBuilder() {
          * Enables or disables creating new accounts in the email sign in flow.
          * <p>
          * <p>Account creation is enabled by default.
+         *
+         * @deprecated set this option directly on the email builder: {@link
+         * IdpConfig.EmailBuilder#setAllowNewAccounts(boolean)}.
          */
         @NonNull
+        @Deprecated
         public SignInIntentBuilder setAllowNewEmailAccounts(boolean enabled) {
             mAllowNewEmailAccounts = enabled;
             return this;
         }
 
+        @NonNull
+        @Override
+        public Intent build() {
+            if (mAllowNewEmailAccounts != null) {
+                // To ensure setAllowNewEmailAccounts backcompat
+                for (IdpConfig provider : mProviders) {
+                    if (provider.getProviderId().equals(EmailAuthProvider.PROVIDER_ID)) {
+                        mProviders.set(mProviders.indexOf(provider), new IdpConfig.EmailBuilder()
```

Comment:
If the loop contains `indexOf` it should probably be a loop over indexes, not a for-each loop.

Reply:
👍